receiver/prometheusreceiver/metrics_receiver_test_helper.go

```diff
@@ -0,0 +1,452 @@
+		t.attributes.Insert("job", pdata.NewAttributeValueString(t.name))
+		t.attributes.Insert("instance", pdata.NewAttributeValueString(u.Host))
+		t.attributes.Insert("port", pdata.NewAttributeValueString(port))
+		t.attributes.Insert("scheme", pdata.NewAttributeValueString("http"))
+	}
+
+	cfgStr := strings.ReplaceAll(string(cfg), srvPlaceHolder, u.Host)
+	pCfg, err := promcfg.Load(cfgStr, false, gokitlog.NewNopLogger())
+	return mp, pCfg, err
+}
+
+func verifyNumScrapeResults(t *testing.T, td *testData, resourceMetrics []*pdata.ResourceMetrics) {
+	want := 0
+	for _, p := range td.pages {
+		if p.code == 200 {
+			want++
+		}
+	}
+	if l := len(resourceMetrics); l != want {
+		t.Fatalf("want %d, but got %d\n", want, l)
+	}
+}
+
+func getMetrics(rm *pdata.ResourceMetrics) []*pdata.Metric {
+	metrics := make([]*pdata.Metric, 0)
+	ilms := rm.InstrumentationLibraryMetrics()
+	for j := 0; j < ilms.Len(); j++ {
+		metricSlice := ilms.At(j).Metrics()
+		for i := 0; i < metricSlice.Len(); i++ {
+			m := metricSlice.At(i)
+			metrics = append(metrics, &m)
+		}
+	}
+	return metrics
+}
+
+func metricsCount(resourceMetric *pdata.ResourceMetrics) int {
+	metricsCount := 0
+	ilms := resourceMetric.InstrumentationLibraryMetrics()
+	for j := 0; j < ilms.Len(); j++ {
+		ilm := ilms.At(j)
+		metricsCount += ilm.Metrics().Len()
+	}
+	return metricsCount
+}
+
+func getValidScrapes(t *testing.T, rms []*pdata.ResourceMetrics) []*pdata.ResourceMetrics {
+	out := make([]*pdata.ResourceMetrics, 0)
+	// rms will include failed scrapes and scrapes that received no metrics but have internal scrape metrics, filter those out
+	for i := 0; i < len(rms); i++ {
+		allMetrics := getMetrics(rms[i])
+		if expectedScrapeMetricCount < len(allMetrics) && countScrapeMetrics(allMetrics) == expectedScrapeMetricCount {
+			if isFirstFailedScrape(allMetrics) {
+				continue
+			}
+			assertUp(t, 1, allMetrics)
+			out = append(out, rms[i])
+		} else {
+			assertUp(t, 0, allMetrics)
+		}
+	}
+	return out
+}
+
+func isFirstFailedScrape(metrics []*pdata.Metric) bool {
+	for _, m := range metrics {
+		if m.Name() == "up" {
+			if m.Gauge().DataPoints().At(0).DoubleVal() == 1 { // assumed up will not have multiple datapoints
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func assertUp(t *testing.T, expected float64, metrics []*pdata.Metric) {
+	for _, m := range metrics {
+		if m.Name() == "up" {
+			assert.Equal(t, expected, m.Gauge().DataPoints().At(0).DoubleVal()) // (assumed up will not have multiple datapoints)
+			return
+		}
+	}
+	t.Error("No 'up' metric found")
+}
+
+func countScrapeMetricsRM(got *pdata.ResourceMetrics) int {
+	n := 0
+	ilms := got.InstrumentationLibraryMetrics()
+	for j := 0; j < ilms.Len(); j++ {
+		ilm := ilms.At(j)
+		for i := 0; i < ilm.Metrics().Len(); i++ {
+			switch ilm.Metrics().At(i).Name() {
+			case "up", "scrape_duration_seconds", "scrape_samples_scraped", "scrape_samples_post_metric_relabeling", "scrape_series_added":
+				n++
+			default:
+			}
```

Comment:
ilm.Metrics().At(i) return metricsSlice object, and countScrapeMetrics() accepts []*pdata.Metrics. Another option was to use this code for countScrapeMetricsRM():
```
n := 0
m := getMetrics(got)
n += countScrapeMetrics(m)
return n: 
```
But in this one metricsSlice will be looped over to convert to []metrics. so to avoid additional loop and to reuse switch statement, pulled the switch into separate method.